scripts/CEE/list-alerts/script.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+# default is retrived all firing alerts (critical+warning)
+IS_CRITICAL=1
+IS_WARNING=1
+PRINT_SILENCE=0
+NAMESPACE_QUERY="namespace=~\"^$|^default$|^openshift-.*|^kube-.*|^redhat-.*\",namespace!~\"^redhat-rhmi-.*\""
+ALERTSTATE_QUERY="alertstate=\"firing\","
+
+for ARG in $(echo "${SCRIPT_PARAMETERS}") ; do
+    if [ "$ARG" == "--warning-only" ] ; then
+        IS_CRITICAL=0
+        IS_WARNING_ONLY=1
+    elif [ "$ARG" == "--critical-only" ] ; then
+        IS_WARNING=0
+        IS_CRITICAL_ONLY=1
+    elif [ "$ARG" == "--pending-only" ] ; then
+        ALERTSTATE_QUERY="alertstate=\"pending\","
+        IS_PENDING_ONLY=1
+    elif [ "$ARG" == "--firing-only" ] ; then
+        ALERTSTATE_QUERY="alertstate=\"firing\","
+        IS_FIRING_ONLY=1
+    elif [ "$ARG" == "--all-states" ] ; then
+        ALERTSTATE_QUERY=""
+        NAMESPACE_QUERY=""
+        IS_ALL_STATES=1
+    elif [ "$ARG" == "--list-silences" ] ; then
+        PRINT_SILENCE=1
+    else
+        echo "usage: script.sh [--warning-only|--critical-only] [--firing-only|--pending-only|--all-states]"
+        echo "  --warning-only: print warning alerts only"
+        echo "  --critical-only: print critical alerts only"
+        echo "  --firing-only: print only firing alert state (default)"
+        echo "  --pending-only: print only pending alert state"
+        echo "  --all-states: print firing and pending alerts"
+        echo "  --list-silences: print all active silences"
+        exit 1
+    fi
+done
+
+function _get_host {
+    oc -n openshift-monitoring get routes "$1" -o json | jq -r .spec.host
+}
+
+if [[ $((IS_FIRING_ONLY+IS_PENDING_ONLY+IS_ALL_STATES)) -gt 1 ]] ; then
+    echo "--pending-only, --firing-only and --all are mutually exclusive."
+    exit 2
+fi
+
+if [[ $((IS_WARNING_ONLY+IS_CRITICAL_ONLY)) -gt 1 ]] ; then
+    echo "--warning-only and --critical-only are mutually exclusive."
+    exit 2
+fi
+
+if [[ $IS_ALL_STATES == 1 ]] ; then
```

Comment:
Could you use the firing-only flag instead?
I like that format more than:
IF all
ELIF one
ELSE two
---
Format I like
IF one
ELIF two
Else all

Reply:
I think it is more logical to have the defaults behavior in the 'else' statement. 
IS_ALL_STATES and IS_PENDING_ONLY are markers for options being used, and the else statement covers 'all other cases' which can be default or usage of the explicit option.

Reply:
I'm good with this as long as the behavior is documented (which is the case)